Source/Helika/Private/HelikaManager.cpp

```diff
@@ -0,0 +1,588 @@
+		return false;
+	}
+
+	// send event to helika API
+	SendHTTPPost("/game/game-event", JsonString);
+
+	return true;
+}
+
+
+bool UHelikaManager::SendUserEvent(TSharedPtr<FJsonObject> EventProps)
+{
+    if(!bIsInitialized)
+    {
+        UE_LOG(LogHelika, Log, TEXT("Helika Subsystem is not yet initialized"));
+    	return false;
+    }
+
+	if(!EventProps.IsValid())
+	{
+		UE_LOG(LogHelika, Error, TEXT("'Event Props' cannot be null"));
+		return false;
+	}
+
+    // adding unique id to event
+    const TSharedPtr<FJsonObject> FinalEvent = MakeShareable(new FJsonObject());
+    FinalEvent->SetStringField("id", FGuid::NewGuid().ToString());   
+
+    TArray<TSharedPtr<FJsonValue>> EventArrayJsonObject;
+    const TSharedPtr<FJsonValueObject> JsonValueObject = MakeShareable(new FJsonValueObject(AppendAttributesToJsonObject(EventProps, true)));
+    EventArrayJsonObject.Add(JsonValueObject);
+    FinalEvent->SetArrayField("events", EventArrayJsonObject);
+
+    // converting Json object to string
+    FString JsonString;
+    const TSharedRef<TJsonWriter<>> Writer = TJsonWriterFactory<>::Create(&JsonString);
+	if(!FJsonSerializer::Serialize(FinalEvent.ToSharedRef(), Writer))
+	{
+		UE_LOG(LogHelika, Error, TEXT("Failed to serialize event data to JSON"));
+		return false;
+	}
+	
+    // send event to helika API
+    SendHTTPPost("/game/game-event", JsonString);
+	return true;
+}
+
+bool UHelikaManager::SendUserEvents(TArray<TSharedPtr<FJsonObject>> EventProps)
+{
+    if(!bIsInitialized)
+    {
+        UE_LOG(LogHelika, Log, TEXT("Helika Subsystem is not yet initialized"));
+    	return false;
+    }
+	
+	if(EventProps.IsEmpty())
+	{
+		UE_LOG(LogHelika, Error, TEXT("'Event Props' cannot be empty"));
+		return false;
+	}
+
+    const TSharedPtr<FJsonObject> FinalEvent = MakeShareable(new FJsonObject());
+    FinalEvent->SetStringField("id", FGuid::NewGuid().ToString());
+    
+    TArray<TSharedPtr<FJsonValue>> EventArrayJsonObject;
+    for (auto EventProp : EventProps)
+    {
+    	if(!EventProp.IsValid())
+    	{
+    		UE_LOG(LogHelika, Error, TEXT("'Event Props' contains invalid/null object"));
+    		return false;
+    	}
+        const TSharedPtr<FJsonValueObject> JsonValueObject = MakeShareable(new FJsonValueObject(AppendAttributesToJsonObject(EventProp, true)));
+        EventArrayJsonObject.Add(JsonValueObject);
+    }        
+    FinalEvent->SetArrayField("events", EventArrayJsonObject);
+    
+    // converting Json object to string
+    FString JsonString;
+    const TSharedRef<TJsonWriter<>> Writer = TJsonWriterFactory<>::Create(&JsonString);
+	if(!FJsonSerializer::Serialize(FinalEvent.ToSharedRef(), Writer))
+	{
+		UE_LOG(LogHelika, Error, TEXT("Failed to serialize event data to JSON"));
+		return false;
+	}
+
+    // send event to helika API
+    SendHTTPPost("/game/game-event", JsonString); 
+
+	return true;
+}
+
+void UHelikaManager::SetPlayerId(const FString& InPlayerId)

```

Comment:
PlayerID is deprecated in favor of UserDetails. Please log an error or warning here saying it's deprecated. Same with GetPlayerID;

Reply:
nevermind, I see the "UE_DEPRECATED(0.1.1, "SetPlayerId() is deprecated. Please use SetUserDetails() instead")" much lower down. 

Should we keep the code in these two functions? or should it just be a no-op? I'm not sure how code deprecation in UE works.